dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java

```diff
@@ -56,32 +60,84 @@
      */
     T getAndCheckResourceByAlias(String alias);
 
-    static void checkAliases(Set<String>  aliases, User user) {
-        // reserve some prefixes for our own use
-        String[] invalidPrefixes = {"dockstore", "doi", "drs", "trs", "dos", "wes"};
+    /**
+     * Check that aliases do not contain invalid prefixes
+     * if the user adding them is not an admin or curator
+     * @param aliases a Set of alias strings
+     * @param user user authenticated to issue a DOI for the workflow
+     * @param blockAliasesWithZenodoFormat block creation of an alias with a particular format
+     * @return the alias as a string
+     */
+    static void checkAliases(Set<String>  aliases, User user, boolean blockAliasesWithZenodoFormat) {
+        // Admins and curators do not have restrictions on alias format
+        if (user.isCurator() || user.getIsAdmin()) {
+            return;
+        }
+
+        checkAliasFormat(aliases, blockAliasesWithZenodoFormat);
+    }
 
+
+    /**
+     * Check that aliases do not contain invalid prefixes
+     * if the user adding them is not an admin or curator
+     * @param aliases a Set of alias strings
+     * @param blockAliasesWithZenodoFormat block creation of an alias with a particular format
+     * @return the alias as a string
+     */
+    static void checkAliasFormat(Set<String>  aliases, boolean blockAliasesWithZenodoFormat) {
         // Gather up any aliases that contain invalid prefixes
         List<String> invalidAliases = new ArrayList<>();
-        if (!user.isCurator() && !user.getIsAdmin()) {
-            invalidAliases = aliases.stream().filter(alias -> StringUtils.startsWithAny(alias, invalidPrefixes)).collect(Collectors.toList());
-        }
+        invalidAliases = aliases.stream().filter(alias -> StringUtils.startsWithAny(alias, INVALID_PREFIXES)).collect(Collectors.toList());
 
         // If there are any aliases with invalid prefixes then report it to the user
         if (invalidAliases.size() > 0) {
             String invalidAliasesString = String.join(", ", invalidAliases);
-            String invalidPrefixesString = String.join(", ", invalidPrefixes);
-            throw new CustomWebApplicationException("These aliases: " + invalidAliasesString + " start with a reserved string,"
+            String invalidPrefixesString = String.join(", ", INVALID_PREFIXES);
+            throw new CustomWebApplicationException("These aliases: " + invalidAliasesString + " start with a reserved string."
                     + " They cannot be used. Please create aliases without these prefixes: " + invalidPrefixesString,
                     HttpStatus.SC_BAD_REQUEST);
         }
+
+        if (blockAliasesWithZenodoFormat) {
+            Pattern pattern = Pattern.compile(ZENDO_DOI_REGEX);
```

Comment:
This will probably get executed so infrequently that it won't effectively matter, but best practice would be to make this a static field, so that the regex is only compiled once instead of every time the method is invoked.

Reply:
done Put pattern outside of method; static not legal for interface